pf4j/src/main/java/org/pf4j/AbstractPluginManager.java

```diff
@@ -275,7 +275,7 @@ protected boolean unloadPlugin(String pluginId, boolean unloadDependents) {
             if (unloadDependents) {
                 List<String> dependents = dependencyResolver.getDependents(pluginId);
                 while (!dependents.isEmpty()) {
-                    String dependent = dependents.remove(0);
+                    String dependent = dependents.get(0);
```

Comment:
I like the change because is very light. 
Did you tested? I ask you because at first glance, this change produces an infinite loop (`dependents` will never be empty and the while block will be executed forever).